packages/nextjs/hooks/scaffold-eth/useDeployedContractInfo.ts

```diff
@@ -1,52 +1,52 @@
-import { useEffect, useState } from "react";
+import { useEffect, useMemo, useState } from "react";
 import { useProvider } from "wagmi";
 import { getTargetNetwork } from "~~/utils/scaffold-eth";
-type GeneratedContractType = {
-  address: string;
-  abi: any[];
-};
+import contracts from "../../generated/hardhat_contracts";
+import { Contract, ContractName, ContractCodeStatus, DefaultChain } from "./contract.types";
 
 /**
  * @dev use this hook to get a deployed contract from `yarn deploy` generated files.
  * @param contractName - name of deployed contract
  * @returns {GeneratedContractType | undefined} object containing contract address and abi or undefined if contract is not found
  */
-export const useDeployedContractInfo = (contractName: string | undefined | null) => {
+export const useDeployedContractInfo = <TContractName extends ContractName>(contractName: TContractName) => {
   const configuredChain = getTargetNetwork();
-  const [deployedContractData, setDeployedContractData] = useState<undefined | GeneratedContractType>(undefined);
-  const [isLoading, setIsLoading] = useState(true);
+  const deployedContract = useMemo(
```

Comment:
it seems you don't need useMemo here

Reply:
I think you're right. Since it's simply accessing an object in a const, it's reference shouldn't change between renders. Well spotted :+1: